### DISCUSSION TITLE
Bulding MoonRay CentOS7 last steps

### DISCUSSION THREAD

**Discussion by xXTR0YXx** [dreamworksanimation/openmoonray#64] [Q&A]
Currently trying to get everything working according to the docs, but the last steps don't seem to work for me

cmake --preset container-release gives me this error: **The source "/openmoonray/CMakeLists.txt" does not match the source "/building/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.**

cmake --build --preset container-release -- -j 64 works and finshes, but have to set the 64 to 32 otherwise my machine chokes

cmake --install /build --prefix /installs/openmoonray gives me this error: **-- Install configuration: ""**

I can already see the finish line, but the last steps are not working for me

**Comment by rgwilson68** [comment]
It looks like the /build directory still contains information from the previous step. Try deleting all the files in /build before running cmake --preset container-release

**Reply by xXTR0YXx** [reply]
yes, now it works until the last point, so the cmake --install /build --prefix /installs/openmoonray doesn't built anything into the /installs/openmoonray folder

**Comment by xXTR0YXx** [comment]
So, everything up and running here. Only thing that gives me errors is when trying to run in XPU mode. 

**Warning (lib.render): GPU: Unable to load PTX file: /installs/openmoonray/scripts/../shaders/GPUPrograms.ptx   ...falling back to CPU vectorized mode
Error: GPU: Unable to load PTX file: /installs/openmoonray/scripts/../shaders/GPUPrograms.ptx   ...falling back to CPU vectorized mode**

**Comment by BrianHanke** [comment]
@xXTR0YXx I had the same problem, solved here: https://github.com/dreamworksanimation/openmoonray/issues/65#issuecomment-1480082936

**Reply by xXTR0YXx** [reply]
Actually it seems to work, just the scene didn't support it. I had other scenes that worked great without complaining and with XPU
[Bildschirmvideo von 2023-03-23 10_49_03 (1).webm](https://user-images.githubusercontent.com/125904224/227200088-597d9ba8-fc01-473e-913a-0937a66dd2d6.webm)

**Comment by xXTR0YXx** [comment]
Any idea why I can't find and run the rdl2_json_exporter? There is no such file in the /bin directory

**Comment by panda1100** [comment]
Hi @xXTR0YXx and  @BrianHanke 

Have you met following problem on your CentOS7 build?
`pxr.MoonrayShaderParser` and `pxr.MoonrayShaderDiscovery` python modules are missing on my CentOS7 build and Rocky Linux 9 build as well.


```
source /installs/openmonray/scripts/setup/sh
export PYTHONPATH=/usr/local/lib/python

hd_render -in /openmoonray/testdata/sphere.usd -out /tmp/sphere.exr

Warning: in Tf_PyLoadScriptModule at line 122 of /build/USD-prefix/src/USD/pxr/base/tf/pyUtils.cpp -- Import failed for module 'pxr.MoonrayShaderParser'!
ModuleNotFoundError: No module named 'pxr.MoonrayShaderParser'
Warning: in Tf_PyLoadScriptModule at line 122 of /build/USD-prefix/src/USD/pxr/base/tf/pyUtils.cpp -- Import failed for module 'pxr.MoonrayShaderDiscovery'!
ModuleNotFoundError: No module named 'pxr.MoonrayShaderDiscovery'
```

**Reply by knightcrawler25** [reply]
The [docs](https://docs.openmoonray.org/user-reference/tools/hydra/hdmoonray-setup/#python-and-hydra-setup) say those warnings are harmless:

![image](https://user-images.githubusercontent.com/11459803/227922691-b8f40312-ebb0-4ebd-8ed5-5969a664bc3f.png)

**Reply by panda1100** [reply]
@knightcrawler25 Thank you for pointing out!! and my apologies I should have carefully read the document..
